nixos/modules/services/networking/pixiecore.nix

```diff
@@ -0,0 +1,133 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pixiecore;
+in
+{
+  meta.maintainers = with maintainers; [ bbigras danderson ];
+
+  options = {
+    services.pixiecore = {
+      enable = mkEnableOption "Pixiecore";
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Open ports (67, 69 UDP and 4011, 'port', 'statusPort' TCP) in the firewall for Pixiecore.
+        '';
+      };
+
+      mode = mkOption {
+        description = "Which mode to use";
+        type = types.enum [ "api" "boot" ];
+      };
+
+      debug = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Log more things that aren't directly related to booting a recognized client";
+      };
+
+      dhcpNoBind = mkOption {
+        type = types.bool;
+        default = false;
```

Comment:
This option does not work because pixiecore does not run as root.
I think this needs some conditional.

Reply:
Maybe there is a capability we can add so listening without binding works.

Reply:
I'm not familiar with listening without binding nor capabilities. Any ideas @danderson ?

Reply:
I just tested it, if you add `cap_net_raw`, when `dhcpNoBind` is set, it works.

```
  AmbientCapabilities = "cap_net_bind_service cap_net_raw";
```